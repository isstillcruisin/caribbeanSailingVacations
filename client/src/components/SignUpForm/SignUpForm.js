import React from "react";
import Card from "react-bootstrap/Card";
import Form from 'react-bootstrap/Form';
import Col from 'react-bootstrap/Col';
import Button from 'react-bootstrap/Button';

const SignUpForm = props => (
   <Card style={{color: 'black'}}>
    <Form>
      <Form.Row>
        <Form.Label column xs={3}>
          Name
        </Form.Label>
        <Col>
          <Form.Control placeholder="First name" name='firstName' onChange={props.handleInputChange} />
        </Col>
        <Col>
          <Form.Control placeholder="Last name" name='lastName' onChange={props.handleInputChange} />
        </Col>
      </Form.Row>
      <Form.Row controlid="formEmail">
        <Form.Label column xs={3}>
          Email
        </Form.Label>
        <Col xs={9}>
          <Form.Control type="email" name='email' placeholder="Email Address" value={props.email} onChange={props.handleInputChange} />
        </Col>
      </Form.Row>
      <Form.Row controlid="formPassword">
        <Form.Label column xs={3}>
          Password
        </Form.Label>
        <Col xs={9}>
          <Form.Control type="password" name='password' placeholder="Choose a password" value={props.password} onChange={props.handleInputChange} />
        </Col>
      </Form.Row>
      <Form.Row controlid="formSubmit">
        <Col>
          <Button
            onClick={props.handleSignUp}
            type="submit"
            className="btn btn-lg"
          >
            Submit
          </Button>
        </Col>
      </Form.Row>
    </Form>
  </Card>
);

export default SignUpForm;
