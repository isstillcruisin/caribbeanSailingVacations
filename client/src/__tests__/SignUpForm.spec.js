import React from 'react'
import { create } from 'react-test-renderer'
import SignUpForm from '../components/SignUpForm'

describe('SignUpForm component', () => {
  test('Matches the snapshot', () => {
    const form = create(<SignUpForm 
      email='test@testerson.com'
      password='aPassword'
    />)
    expect(form.toJSON()).toMatchSnapshot()
  })
})