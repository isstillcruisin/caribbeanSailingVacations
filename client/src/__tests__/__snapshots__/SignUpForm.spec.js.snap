// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm component Matches the snapshot 1`] = `
<div
  className="card"
  style={
    Object {
      "color": "black",
    }
  }
>
  <form
    className=""
  >
    <div
      className="form-row"
    >
      <label
        className="form-label col-form-label col-3"
      >
        Name
      </label>
      <div
        className="col"
      >
        <input
          className="form-control"
          name="firstName"
          placeholder="First name"
        />
      </div>
      <div
        className="col"
      >
        <input
          className="form-control"
          name="lastName"
          placeholder="Last name"
        />
      </div>
    </div>
    <div
      className="form-row"
      controlid="formEmail"
    >
      <label
        className="form-label col-form-label col-3"
      >
        Email
      </label>
      <div
        className="col-9"
      >
        <input
          className="form-control"
          name="email"
          placeholder="Email Address"
          type="email"
          value="test@testerson.com"
        />
      </div>
    </div>
    <div
      className="form-row"
      controlid="formPassword"
    >
      <label
        className="form-label col-form-label col-3"
      >
        Password
      </label>
      <div
        className="col-9"
      >
        <input
          className="form-control"
          name="password"
          placeholder="Choose a password"
          type="password"
          value="aPassword"
        />
      </div>
    </div>
    <div
      className="form-row"
      controlid="formSubmit"
    >
      <div
        className="col"
      >
        <button
          className="btn btn-lg btn btn-primary"
          disabled={false}
          type="submit"
        >
          Submit
        </button>
      </div>
    </div>
  </form>
</div>
`;
